core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1005,23 +1056,171 @@ private void processBind(@Transient ObjList<BindVariableSetter> bindVariableSett
+            }
+            bindVariableTypes.clear();
+            setupBindVariables(lo, parameterCount, bindVariableSetters, bindVariableTypes);
+        } else if (parameterCount < 0) {
+            LOG.error()
+                    .$("invalid parameter count [parameterCount=").$(parameterCount)
+                    .$(", offset=").$(lo - address)
+                    .$(']').$();
+            throw BadProtocolException.INSTANCE;
+        }
+
+        // at this point we may have a current query that is not null
+        // this is ok to lose reference to this query because we have cache
+        // of all of them, which is looked up by query text
+        responseAsciiSink.reset();
+        final Object statement = factoryCache.peek(queryText);
+        if (statement == null) {
+            if (queryText.length() > 0) {
+                final CompiledQuery cc = compiler.compile(queryText, sqlExecutionContext);
+                sqlExecutionContext.storeTelemetry(cc.getType(), TelemetryOrigin.PG_WIRE);
+
+                switch (cc.getType()) {
+                    case CompiledQuery.SELECT:
+                        currentFactory = cc.getRecordCursorFactory();
+                        queryTag = TAG_SELECT;
+                        factoryCache.put(queryText, currentFactory);
+                        break;
+                    case CompiledQuery.INSERT:
+                        currentInsertStatement = cc.getInsertStatement();
+                        queryTag = TAG_INSERT;
+                        factoryCache.put(queryText, currentInsertStatement);
+                        break;
+                    case CompiledQuery.COPY_LOCAL:
+                        queryTag = TAG_COPY;
+                        sendCopyInResponse(compiler.getEngine(), cc.getTextLoader());
+                        break;
+                    case CompiledQuery.SET:
+                        queryTag = TAG_SET;
+                        break;
+                    default:
+                        // DDL SQL
+                        queryTag = TAG_OK;
+                        break;
+                }
+            } else {
+                isEmptyQuery = true;
+            }
+        } else {
+            if (statement instanceof RecordCursorFactory) {
+                queryTag = TAG_SELECT;
+                currentFactory = (RecordCursorFactory) statement;
+            } else if (statement instanceof InsertStatement) {
+                queryTag = TAG_INSERT;
+                currentInsertStatement = (InsertStatement) statement;
+            } else {
+                assert false;
+            }
+        }
+        //cache named statement
+        if (statementName != null) {
+            NamedStatementWrapper wrapper = namedStatementWrapperStack.pop();
+            if (wrapper == null) {
+                wrapper = new NamedStatementWrapper();
+            }
+            if (currentFactory != null) {
+                wrapper.selectFactory = currentFactory;
+            }
```

Comment:
factory and insert statement are mutually exclusive, we should have `else if` here